src/in-view.js

```diff
@@ -58,69 +66,76 @@ const inView = () => {
     * The main interface. Take a selector and retrieve
     * the associated registry or create a new one.
     */
-    let control = (selector) => {
-
-        if (typeof selector !== 'string') return;
-
-        // Get an up-to-date list of elements.
-        let elements = [].slice.call(document.querySelectorAll(selector));
+    let control = (selector, opts) => {
+
+        let elements;
+
+        if (typeof selector === 'string') {
+            // Get an up-to-date list of elements.
+            elements = [].slice.call(document.querySelectorAll(selector));
+        } else {
+            if (selector instanceof window.Node) {
+                elements = [selector];
+            } else if (selector instanceof window.NodeList || selector instanceof Array) {
+                elements = [].slice.call(selector);
+            } else {
+                // selector is not supported
+                return;
+            }
+            // create unique selector
+            selector = 'Node-'+nodeCounter++;
```

Comment:
`nodeCounter` is defined on line 30 as an array, what does this statement do?

Reply:
good catch

Reply:
@roberthl fixed